src/RouterMiddleware.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Idiosyncratic\Http\Router;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Throwable;
+
+class RouterMiddleware implements MiddlewareInterface
+{
+    /** @var Router */
+    private $router;
+
+    public function __construct(Router $router)
+    {
+        $this->router = $router;
+    }
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler) : ResponseInterface
+    {
+        try {
+            return $this->router->dispatch($request);
+        } catch (Throwable $e) {
+            return $handler->handle($request);
```

Comment:
Wouldn't it make sense for your `Router` implementation be the `RequestHandlerInterface` implementation?

Reply:
>If an error is produced, why are you "handling" the request? I could maybe understand something like NoMatchingRouteException being caught and falling back to a default request handler, but as it is, this doesn't make sense to me. What am I missing?

In the case of a middleware context, this would not be the last middleware to run - `$handler` is the next middleware in the stack.

>Wouldn't it make sense for your Router implementation be the RequestHandlerInterface implementation?

I'm next going to add a separate `RequestHandlerInterface` implementation that I think would be behave more in line with your expectations here.

Reply:
> In the case of a middleware context, this would not be the last middleware to run - $handler is the next middleware in the stack.

That makes more sense, but I'm failing to see how you would recover from a fatal error from this middleware.  I _could_ see something like this being viable:
```php
class ErrorHandlingMiddleware implements MiddlewareInterface
{
    ...
    public function process(...)
    {
        try {
            $handler->handle($request);
        } catch (Throwable $t) {
            return $this->responseFactory->fromError($t);
        }
    }
}
```

Reply:
I think I need to read the spec more to understand this.  I just found this, for instance.

>1.4 Handling Exceptions
>
>It is RECOMMENDED that any application using middleware includes a component that catches exceptions and converts them into responses. This middleware SHOULD be the first component executed and wrap all further processing to ensure that a response is always generated.

Reply:
I'll be honest, I glossed over section 1.4 when reading through the spec (which is a sad commentary, given how short the spec is). I just assumed that a Middleware should do everything it possibly can to not throw any exceptions, given that in `1.1 Request Handlers` it specifically states:

> A request handler MAY throw an exception if request conditions prevent it from producing a response.

and there's not a corresponding statement in `Section 1.2 Middleware`. I may have to ponder this some more.

I like the `Throwable $t` in your example.  Out of habit I still always use either `$e` or `$exception` (whether I abbreviate largely depends on my mood at the time) as the variable for catch blocks, even since switching to type hinting with `Throwable`.